alpine/fetcher.go

```diff
@@ -58,18 +58,16 @@ func (u *Updater) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCl
 
 	var r io.Reader = res.Body
 	if _, err := io.Copy(tf, r); err != nil {
+		tf.Close()
```

Comment:
```suggestion
		if tf != nil {
		  tf.Close()
		}
```

I've gotten into the habit of checking nil of these things first, may not be necessary.

Reply:
In this case, tmp.File works such that if it didn't return an error, it can't return a nil pointer.